examples/keras_io/tensorflow/vision/mobilevit.py

```diff
@@ -0,0 +1,380 @@
+        label = tf.one_hot(label, depth=num_classes)
+        return image, label
+
+    return _pp
+
+
+def prepare_dataset(dataset, is_training=True):
+    if is_training:
+        dataset = dataset.shuffle(batch_size * 10)
+    dataset = dataset.map(preprocess_dataset(is_training), num_parallel_calls=auto)
+    return dataset.batch(batch_size).prefetch(auto)
+
+
+"""
+The authors use a multi-scale data sampler to help the model learn representations of
+varied scales. In this example, we discard this part.
+"""
+
+"""
+## Load and prepare the dataset
+"""
+
+train_dataset, val_dataset = tfds.load(
+    "tf_flowers", split=["train[:90%]", "train[90%:]"], as_supervised=True
+)
+
+num_train = train_dataset.cardinality()
+num_val = val_dataset.cardinality()
+print(f"Number of training examples: {num_train}")
+print(f"Number of validation examples: {num_val}")
+
+train_dataset = prepare_dataset(train_dataset, is_training=True)
+val_dataset = prepare_dataset(val_dataset, is_training=False)
+
+"""
+## Train a MobileViT (XXS) model
+"""
+
+learning_rate = 0.002
+label_smoothing_factor = 0.1
+epochs = 30
+
+optimizer = keras.optimizers.Adam(learning_rate=learning_rate)
+loss_fn = keras.losses.CategoricalCrossentropy(label_smoothing=label_smoothing_factor)
+
+
+def run_experiment(epochs=epochs):
+    mobilevit_xxs = create_mobilevit(num_classes=num_classes)
+    mobilevit_xxs.compile(optimizer=optimizer, loss=loss_fn, metrics=["accuracy"])
+
+    checkpoint_filepath = "/tmp/checkpoint.weights.h5"
+    checkpoint_callback = keras.callbacks.ModelCheckpoint(
+        checkpoint_filepath,
+        monitor="val_accuracy",
+        save_best_only=True,
+        save_weights_only=True,
+    )
+
+    mobilevit_xxs.fit(
+        train_dataset,
+        validation_data=val_dataset,
+        epochs=epochs,
+        callbacks=[checkpoint_callback],
+    )
+    mobilevit_xxs.load_weights(checkpoint_filepath)
+    _, accuracy = mobilevit_xxs.evaluate(val_dataset)
+    print(f"Validation accuracy: {round(accuracy * 100, 2)}%")
+    return mobilevit_xxs
+
+
+mobilevit_xxs = run_experiment()
+
+"""
+## Results and TFLite conversion
+
+With about one million parameters, getting to ~85% top-1 accuracy on 256x256 resolution is
+a strong result. This MobileViT mobile is fully compatible with TensorFlow Lite (TFLite)
+and can be converted with the following code:
+"""
+
+# Serialize the model as a SavedModel.
+tf.saved_model.save(mobilevit_xxs, "mobilevit_xxs")
+
+# Convert to TFLite. This form of quantization is called
+# post-training dynamic-range quantization in TFLite.
+converter = tf.lite.TFLiteConverter.from_saved_model("mobilevit_xxs")
+converter.optimizations = [tf.lite.Optimize.DEFAULT]
+converter.target_spec.supported_ops = [
+    tf.lite.OpsSet.TFLITE_BUILTINS,  # Enable TensorFlow Lite ops.
+    tf.lite.OpsSet.SELECT_TF_OPS,  # Enable TensorFlow ops.
+]
+tflite_model = converter.convert()
+open("mobilevit_xxs.tflite", "wb").write(tflite_model)
```

Comment:
Did you check that the TFLite model works?

Reply:
I didn't perform the inference using TFLite model, but I verified that this line is saving TFLite model.

Reply:
Can you try using the TFLite interpreter to check that it works as expected? https://www.tensorflow.org/lite/guide/inference#load_and_run_a_model_in_python